### PR TITLE
USWDS - Print styles integration into code base

### PR DESCRIPTION
**Pull request by mitchmoccia**
This adds the base print styles file to 
src/stylesheets/base/_print.scss

Also, the theme custom styles file
src/stylesheets/theme/_uswds-theme-custom-styles.scss
contains a couple of commented out snippets that instruct the developer to add their own custom styles to it, and also addresses the use of absolute paths not displaying the domain name and offers a fix for that if they choose.

Fed link for reference:
https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/accelerator/3940-printer-friendly/

Note: Print styles doesn't appear in the left navigation but it is ever present. Go here and do a print preview:
https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/accelerator/3940-printer-friendly/components/preview/layout--create-account.html

Related PR for guidance:
https://github.com/uswds/uswds-site/pull/1158

### CODE REVIEW

**Review by mejiaj — COMMENTED** [review]
There might be some changes we can make to add clarity to some components. There are also 2 hex colors we're using where we could potentially use USWDS colors.

File: src/stylesheets/base/_print.scss [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  // plus remove any shadow effects
+  *,
+  *:before,
+  *:after,
+  *:first-letter,
+  p:first-line,
+  div:first-line,
+  blockquote:first-line,
+  li:first-line {
+    background: transparent !important;
```
Comment (mejiaj) [inline src/stylesheets/base/_print.scss:12]: This removes icons on form controls.

File: src/stylesheets/base/_print.scss [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    color: #000 !important;
+    box-shadow: none !important;
```
Comment (mejiaj) [inline src/stylesheets/base/_print.scss:14]: This removes outlines on all buttons.

File: src/stylesheets/base/_print.scss [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  article {
+    width: 100% !important;
+    margin: 0 !important;
+    padding: 0 !important;
+  }
+  // Hide elements that are not necessary for print
+  nav,
+  aside,
+  footer {
+    display: none !important;
```
Comment (mejiaj) [inline src/stylesheets/base/_print.scss:28]: This completely hides breadcrumbs and eventually pagination on print.

image

File: src/stylesheets/base/_print.scss [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  a[href^="javascript:"]:after {
+    content: "" !important;
+  }
+  pre {
+    white-space: pre-wrap !important;
+  }
+  // Add a border around pre and blockquote tags to clearly indicate these elements in print
+  pre,
+  blockquote {
+    border: 1px solid #999 !important;
```
Comment (mejiaj) [inline src/stylesheets/base/_print.scss:59]: Should we use a color token here?

**Comment by maya** [comment]
If you'd like to test this on a site that needs printing, feel free to use https://simplereport.gov/resources/ - I tested adding the print styles in this PR to the site but didn't see any change to the print preview.

**Comment by thisisdano** [comment]
This is going to need some more work before it's ready to release — I'm going to close this until we have time to pick it up again. When we do pick it up again, I think it'll need a different approach.

**Pull request closed without merging**